### ISSUE TITLE
Void Payment Webhook Doesn't Match Recurly Documentation

### ISSUE THREAD

**Issue by jkeam** [killbilling/recurly-java-library#75]
The void payment webhook doesn't parse the example in the recurly documentation.  It's a real simple fix.  The name of your class, and the string that you are expecting is "VoidedPaymentNotification.java", or "voided_payment_notification".  However, as per their documentation, they are saying it's void_payment_notification.  I don't know how to trigger recurly to send a void payment webhook, and am not really sure who is right, the documentation or your code, but either way, I wanted to open an issue and document it.

https://docs.recurly.com/api/push-notifications

```xml
<?xml version="1.0" encoding="UTF-8"?>
<void_payment_notification>
  <account>
    <account_code>1</account_code>
    <username nil="true"></username>
    <email>verena@example.com</email>
    <first_name>Verena</first_name>
    <last_name>Example</last_name>
    <company_name nil="true"></company_name>
  </account>
  <transaction>
    <id>4997ace0f57341adb3e857f9f7d15de8</id>
    <invoice_id>ffc64d71d4b5404e93f13aac9c63b007</invoice_id>
    <invoice_number_prefix></invoice_number_prefix>
    <invoice_number type="integer">2059</invoice_number>
    <subscription_id>1974a098jhlkjasdfljkha898326881c</subscription_id>
    <action>purchase</action>
    <date type="datetime">2010-10-05T23:00:50Z</date>
    <amount_in_cents type="integer">235</amount_in_cents>
    <status>void</status>
    <message>Test Gateway: Successful test transaction</message>
    <reference></reference>
    <source>subscription</source>
    <cvv_result code="M">Match</cvv_result>
    <avs_result code="D">Street address and postal code match.</avs_result>
    <avs_result_street></avs_result_street>
    <avs_result_postal></avs_result_postal>
    <test type="boolean">true</test>
    <voidable type="boolean">false</voidable>
    <refundable type="boolean">false</refundable>
  </transaction>
</void_payment_notification>
```

**Comment by cgerrior** [comment]
Hi @jkeam,

Thanks for reporting this. I can confirm that it is "void_payment_notification"...

<?xml version="1.0" encoding="UTF-8"?>
<void_payment_notification>
  <account>
    <account_code>1877359</account_code>
    <username nil="true"></username>
    <email>maranda.williamson@example.com</email>
    <first_name>Alfreda</first_name>
    <last_name>Howe</last_name>
    <company_name nil="true"></company_name>
  </account>
  <transaction>
    <id>30071b3774f7b0c16829ba4607926680</id>
    <invoice_id>30071b37577866d8e63a8240798c8430</invoice_id>
    <invoice_number_prefix></invoice_number_prefix>
    <invoice_number type="integer">14885</invoice_number>
    <subscription_id>30071b372496b4411141c64eb6ab800c</subscription_id>
    <action>purchase</action>
    <date type="datetime">2015-07-16T19:20:54Z</date>
    <amount_in_cents type="integer">1000</amount_in_cents>
    <status>void</status>
    <message>Successful test transaction</message>
    <reference>4530042</reference>
    <source>subscription</source>
    <cvv_result code=""></cvv_result>
    <avs_result code="D">Street address and postal code match.</avs_result>
    <avs_result_street nil="true"></avs_result_street>
    <avs_result_postal nil="true"></avs_result_postal>
    <test type="boolean">true</test>
    <voidable type="boolean">false</voidable>
    <refundable type="boolean">false</refundable>
  </transaction>
</void_payment_notification>

**Comment by jkeam** [comment]
Thanks @cgerrior for the verification!  I opened a PR just now against this issue that should resolve this.  The XMLRootElement was good on the void payment object, but the real problem was in the Type enum found in Notification.java.  The enum was incorrect, so Notification couldn't properly detect what type it was.  I fixed that and just went ahead and refactored the name of the object to match what Recurly is calling it, VoidPaymentNotification.  I also added a test, TestVoidNotification that demonstrates the issue.
